src/main/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImport.java

```diff
@@ -0,0 +1,343 @@
+
+        try {
+            logger.info("GenomicsDB import batch started");
+            importer = new GenomicsDBImporter(sampleToVCFMap, mergedHeader, intervals.get(0), workspace,
+                                              arrayName, variantContextBufferSize, segmentSize,
+                                              vidMapJSONFile.getAbsolutePath(),
+                                              callsetMapJSONFile.getAbsolutePath());
+        } catch (IOException e) {
+            throw new UserException("GenomicsDB constructor has errors: " + e.getMessage());
+        }
+    }
+
+    /**
+     * Input argument "overwriteExistingWorkspace" defaults to false.
+     * The tool should create a new workspace by default if the workspace
+     * doesn't already exist, but only overwrite an existing workspace
+     * if the user specifies this argument.
+     *
+     * @return  The workspace directory
+     */
+    private File checkOrCreateWorkspace() {
+        File workspaceDir = new File(workspace);
+
+        if (overwriteExistingWorkspace) {
+            IOUtils.tryDelete(workspaceDir);
+        }
+
+        int ret = GenomicsDBImporter.createTileDBWorkspace(workspaceDir.getAbsolutePath());
+
+        if (ret > 0) {
+            File tempFile = new File(workspaceDir.getAbsolutePath() + "/__tiledb_workspace.tdb");
+            if (!tempFile.exists()) {
+                throw new UserException(workspaceDir.getAbsolutePath() + " is not a valid GenomicsDB workspace");
+            }
+
+            logger.info("Importing data to GenomicsDB workspace: " + workspaceDir);
+        } else if (ret < 0) {
+            throw new UserException("Error creating GenomicsDB workspace: " + workspaceDir);
+        }
+
+        return workspaceDir;
+    }
+
+    /**
+     * A complete traversal from start to finish. This method will import all samples
+     * specified in the input GVCF files.
+     */
+    @Override
+    public void traverse() {
+        try {
+            importer.importBatch();
+        } catch (IOException e) {
+            throw new UserException("GenomicsDB importBatch raised exception: " + e.getMessage());
+        }
+    }
+
+    @Override
+    public Object onTraversalSuccess() {
+        logger.info("Import completed!");
+        return importer.isDone();
+    }
+
+    @Override
+    public void closeTool() {
+        for (Map.Entry<String, FeatureReader<VariantContext>> reader : sampleToVCFMap.entrySet()) {
+            try {
+               reader.getValue().close();
+            } catch (IOException e) {
+                throw new GATKException("FeatureReader close() for " + reader.getKey() +
+                    " has thrown an IOException: " + e.getMessage());
+            }
+        }
+    }
+
+    /**
+     * Before traversal starts, create the feature readers
+     * for all the input GVCFs, create the merged header and
+     * initialize the interval
+     */
+    @Override
+    public void onStartup() {
```

Comment:
Move `onStartup() to the top, just before `onTraversalStart()`, for readability's sake